src/Symfony/Component/Messenger/Worker.php

```diff
@@ -65,11 +68,19 @@ public function run(array $options = []): void
         $options = array_merge([
             'sleep' => 1000000,
         ], $options);
+        $queueNames = $options['queues'] ?? false;
 
         while (false === $this->shouldStop) {
             $envelopeHandled = false;
             foreach ($this->receivers as $transportName => $receiver) {
-                $envelopes = $receiver->get();
+                if ($queueNames) {
+                    if (!$receiver instanceof QueueReceiverInterface) {
```

Comment:
we do this check too often i think, but not sure if it matters. putting it outside of this loop is more code, but maybe also more readable.

but this is a detail, lets decide if we like the approach before going into implementation details.